src/ur/workday.ur

```diff
@@ -0,0 +1,378 @@
+}
+val _ : json attachment =
+    json_record {Id = "id",
+                 Description = "description",
+                 ContentType = "contentType",
+                 FileLength = "fileLength",
+                 UploadDate = "uploadDate",
+                 FileName = "fileName",
+                 UploadedBy = "uploadedBy",
+                 Category = "category"}
+
+type businessProcessParameters = {
+     For : option descriptor, (* could be another business process if this is a subprocess *)
+     TransactionStatus : option descriptor,
+     Action : option descriptor,
+     OverallBusinessProcess : option descriptor,
+     Comment : option string,
+     Comments : option (list comment),
+     WarningValidations : option string,
+     Attachments : option (list attachment),
+     OverallStatus : option string,
+     CriticalValidations : option string,
+}
+val _ : json businessProcessParameters =
+    json_record_withOptional
+        {}
+        {For = "for",
+         Action = "action",
+         TransactionStatus = "transactionStatus",
+         OverallBusinessProcess = "overallBusinessProcess",
+         Comment = "comment",
+         Comments = "comments",
+         WarningValidations = "warningValidations",
+         Attachments = "attachments",
+         OverallStatus = "overallStatus",
+         CriticalValidations = "criticalValidations"}
+
+(* Individual response to feedback questions. *)
+type feedbackResponse = {
+     Id : wid,
+     Response : string,
+     Declined : bool,
+     DeclineReason : string,
+     Responder : descriptor,
+     Preview : string
+}
+val _ : json feedbackResponse =
+    json_record {Id = "id",
+                 Response = "response",
+                 Declined = "feedbackDeclined",
+                 DeclineReason = "feedbackDeclineReason",
+                 Responder = "feedbackResponder",
+                 Preview = "descriptor"}
+
+type feedbackQuestion = {
+     Id : wid,
+     Question : string,
+     Responses : list feedbackResponse,
+     QuestionType : descriptor,
+     RelatesTo : list descriptor,
+     MultipleChoiceAnswers : list descriptor,
+     Preview : string
+}
+val _ : json feedbackQuestion =
+    json_record {Id = "id",
+                 Question = "question",
+                 Responses = "feedbackResponses",
+                 QuestionType = "questionType",
+                 RelatesTo = "relatesTo",
+                 MultipleChoiceAnswers = "questionMultipleChoiceAnswers",
+                 Preview = "descriptor"}
+
+type relatedFeedback = {
+     Id : wid,
+     BusinessProcessParameters : businessProcessParameters,
+     Preview : string
+}
+val _ : json relatedFeedback =
+    json_record {Id = "id",
+                 Preview = "descriptor",
+                 BusinessProcessParameters = "businessProcessParameters"}
+
+type anytimeFeedback = {
+     Id : wid,
+     Preview : option string,
+     ToWorker : option descriptor, (* who the feedback is about *)
```

Comment:
Because this code is duplicated from the `.urs` file, I would usually leave out comments here, since they must make more work to keep the copies in-sync.  Conversely, it is probably good to add more comments in the `.urs` file, describing record fields.

Reply:
@achlipala-biz That's sensible, but funny enough, I've taken a different approach.  When I have to change a public type, I make the change in the urs file and then copy the definition wholesale into the ur file, which ends up feeling like less work so long as the two definitions are identical (down to comments).

Some day we should put together a real, actual style guide for urweb and pick one of these.